integrations/hardhat/index.js

```diff
@@ -0,0 +1,9 @@
+const sapphire = require('@oasisprotocol/sapphire-paratime');
+const { extendEnvironment } = require('hardhat/config');
+
+extendEnvironment((hre) => {
+  const { chainId, url: rpcUrl } = hre.network.config;
+  const isSapphire = sapphire.NETWORKS[chainId] || /sapphire/i.test(rpcUrl);
```

Comment:
would you prefer something like

a:
```ts
const isSapphire = chainId ? !!sapphire.NETWORKS[chainId] : /sapphire/i.test(rpcUrl);
```

or b:
```ts
if (!chainId && /sapphire/i.test(rpcUrl)) {
    console.warn('network with `rpcUrl`, rpcUrl, 'may be Sapphire, but cannot be wrapped by `@oasisprotocol/sapphire-hardhat` because you didn't specify a `chainId`');
    return;
}
```

or c:
```
some other option
```

Reply:
It's risky if an un-encrypted transaction gets sent to the web3 gateway. It's annoying if the plugin is over-zealous.